### ISSUE TITLE
Support `--build` compiler option for typescript projects

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/stryker#1334]
We're currently not support the `--build` compiler option for typescript projects. 

Reproduction steps:

```shell
git clone https://github.com/nicojs/typed-inject/
cd typed-inject
npm i
npx stryker run
```

**Comment by nicojs** [comment]
Apparently, the TS team didn't open the compiler API for the `--build` option yet:

https://github.com/Microsoft/TypeScript/blob/fa74cef81e488c8e5bf790020620f0226118cb58/src/compiler/tsbuild.ts#L2

@sheetalkamat Is it OK to ask a question? Since you were the one to add the `@internal` flag. Is there a timeline for this? We would love to support projects being build with `--build`.

**Comment by nicojs** [comment]
Interesting issues:
* https://github.com/Microsoft/TypeScript/issues/28640
* https://github.com/TypeStrong/ts-loader/issues/851

**Comment by Djaler** [comment]
Hey, @nicojs. Looks like this was implemented - https://github.com/microsoft/TypeScript/pull/31432

**Comment by nicojs** [comment]
Yep, looks like it. Not sure if it is already released though, I don't think so.

**Comment by nicojs** [comment]
It is released in 3.6: https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/#apis-to-support-build-and-incremental

We can base our implementation on ts-loader: https://github.com/TypeStrong/ts-loader/pull/935
